File: 프로그래머스/0/181930. 주사위 게임 2/주사위 게임 2.java
import java.util.Set;
import java.util.stream.Stream;
import java.util.stream.Collectors;

class Solution {
    public int solution(int a, int b, int c) {
        Set<Integer> set = Stream.of(a, b, c).collect(Collectors.toSet());
        
        if (set.size() == 3) {
            return a + b + c;
        } else if (set.size() == 2) {
            return (a + b + c) * (int)(Math.pow(a, 2) + Math.pow(b, 2) + Math.pow(c, 2));
        }
        return (a + b + c) * (int)(Math.pow(a, 2) + Math.pow(b, 2) + Math.pow(c, 2)) * (int)(Math.pow(a, 3) + Math.pow(b, 3) + Math.pow(c, 3));
    }
}

File: 프로그래머스/0/181930. 주사위 게임 2/주사위 게임 2.cpp
#include <string>
#include <vector>
#include <math.h>
#include <set>

using namespace std;

int solution(int a, int b, int c) {
    set<int> s = {a, b, c};
    if (s.size() == 3) {
        return a + b + c;
    } else if (s.size() == 2) {
        return (a + b + c) * (pow(a, 2) + pow(b, 2) + pow(c, 2));
    } else {
        return (a + b + c) * (pow(a, 2) + pow(b, 2) + pow(c, 2)) * (pow(a, 3) + pow(b, 3) + pow(c, 3));
    }
}